src/layout/encode.ts

```diff
@@ -66,7 +67,12 @@ export default async function(type: Event): Promise<void> {
                 }
                 tokens = tokenize(tokens, metadata);
             }
-            queue(tokens);
+
+            // Layout events are queued based on the active configuration
+            // If lean mode is on, instead of sending these events to server, we back them up in memory using Event.Backup.
```

Comment:
Will we limit the amount of data queued up for this? Seems like it could grow too large if we don't cap it. Maybe cap at a max size of X KB.

Reply:
This is tricky; let's talk later today about this. It was a concern in my mind too but if we pause this data, or do FIFO - the data quality won't hold anymore. Leaving this open for us to discuss later. 

---
In reply to: [369063178](https://github.com/microsoft/clarity-js/pull/166#discussion_r369063178) [](ancestors = 369063178)